### ISSUE TITLE
Improve testing coverage

### ISSUE THREAD

**Issue by shlomi-nx** [nutanix/papiea-js#60]
Some things are not covered at all (at least in the report), such as `entity`, `provider` and `validator`.

**Comment by Emulebest** [comment]
@shlomi-nx @joshua-berry-ntnx I don't think this is relevant anymore, we pretty much have coverage for everything. We might want to improve coverage on some aspects but IMO this is not a milestone-ish issue if we don't have any specific problematic area that causes a lot of troubles due to lack of testing

**Comment by joshua-berry-ntnx** [comment]
I'm OK to close and/or split this, up to you guys.
